compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1295,9 +1301,35 @@ pub fn catch_with_exit_code(f: impl FnOnce() -> interface::Result<()>) -> i32 {
     }
 }
 
-/// Stores the default panic hook, from before [`install_ice_hook`] was called.
-static DEFAULT_HOOK: OnceLock<Box<dyn Fn(&panic::PanicInfo<'_>) + Sync + Send + 'static>> =
-    OnceLock::new();
+pub static ICE_PATH: OnceLock<Option<PathBuf>> = OnceLock::new();
+
+pub fn ice_path() -> &'static Option<PathBuf> {
+    ICE_PATH.get_or_init(|| {
+        if !rustc_feature::UnstableFeatures::from_environment(None).is_nightly_build() {
+            return None;
+        }
+        if let Ok("0") = std::env::var("RUST_BACKTRACE").as_deref() {
+            return None;
+        }
+        let mut path = match std::env::var("RUSTC_ICE").as_deref() {
+            // Explicitly opting out of writing ICEs to disk.
+            Ok("0") => return None,
+            Ok(s) => match PathBuf::try_from(s) {
+                Ok(p) => p,
+                // On parse error, fallback to the default directory, otherwise the backtraces might be
```

Comment:
If you use `var_os`  the only failure you can get is the env var not being set

Reply:
Huh, this is PathBuf::try_from, isn't that infallible?

Reply:
Yea something is fishy, please remove the two error paths, they are unreachable afaict

Reply:
Leaving the top level match because we need to account for the env var not being set.